src/workerd/util/sqlite.c++

```diff
@@ -531,11 +531,11 @@ void SqliteDatabase::setupSecurity() {
   // We use the suggested limits from the web site. Note that sqlite3_limit() does NOT return an
   // error code; it returns the old limit.
   sqlite3_limit(db, SQLITE_LIMIT_LENGTH, 1000000);
-  sqlite3_limit(db, SQLITE_LIMIT_SQL_LENGTH, 100000);
+  sqlite3_limit(db, SQLITE_LIMIT_SQL_LENGTH, 1000000);
```

Comment:
As discussed I would prefer to relax only the limits that we know we need to relax for some use case.

The SQLite security page recommended these limits presumably for a reason, so we shouldn't relax them without thinking carefully.

Reply:
Oh, I missed your explanations in the PR description.

However, it seems like a lot of this is driven by the desire to import data by specifying it literally in SQL statements -- i.e. not using paramaters. For example, `SQLITE_LIMIT_SQL_LENGTH` is a limit on SQL statement length -- it does _not_ limit the size of the data, unless the data is being specified literally in SQL.

Specifying data literally is something we should be discouraging, not just because of SQL injection but also because it's a lot less efficient. If you want to insert 1000 rows into a table, compiling them as one big sqlite statement that contains all the values as literals is going to be way less efficient than compiling a single-row parameterized insert statement and calling it 1000 times. In general, one-off sql statements are slow.

That said, I suppose the driving motivation here is being able to import sqlite dumps? I wonder if there's a better way for us to support that specifically...

Reply:
Yeah for D1, `wrangler d1 execute my-db --file ./data.sql` and [migrations](https://developers.cloudflare.com/d1/platform/migrations/) are the ways you ingest data. But wrangler and the D1 API just turn them into `.batch` calls, which get executed within the D1's durable object like any other query.

I suppose we could drop the `SQLITE_LIMIT_LENGTH` to 100k so that all the limits were at least consistent with each other? And then if we choose to raise them, raise them all together?

Alternatively, we'd need to treat "SQL file ingestion" as a separate operation that wasn't subject to these limits. That might be kinda nice anyway, since we struggle to handle large SQL dumps at the moment.